### ISSUE TITLE
Error in "Node" %in% class(node) :    is_scalar_character(total_lab) is not TRUE

### ISSUE THREAD

**Issue by sarah872** [uRosConf/voronoiTreemap#5]
Hi there,
first of all, thanks for the wonderful tool!
I tried to use it with my own dataset, but I am getting the following error:
```
Error in "Node" %in% class(node) : 
  is_scalar_character(total_lab) is not TRUE
```
The dataframe looks like this:
```
Proteobacteria Alphaproteobacteria Parvularculales   #FCCDE5   1.04 Parvularculales  
Proteobacteria Oligoflexia Bdellovibrionales       #FCCEE5   3.12 Bdellovibrionales  
[...]
```
I tried to format it as your example file (canada).
Column 1 to 4 & 6 are character columns (called 'h1' 'h2' 'h3' 'color' 'codes'). Column 5 is numeric (called 'weight') and contains a percent value (the sum of this column is 100).

At the very first step:
`canadaH <- vt_export_json(vt_input_from_df(data_sum,scaleToPerc = F))`
I get the error. Any idea why?

**Comment by alexkowa** [comment]
Hi Sarah,
I don't see the problem immediately. 
But we had the improvement of `vt_input_from_df` on our radar anyway, so I added new parameters to it, where you have to explicitly name the columns containing the information. So if you install the github development version `devtools::install_github("uRosConf/voronoiTreemap")` and try the new version of the function `vt_input_from_df` it would be great.

If the problem still exists, can you provide me with a dummy data set and the code you use.

**Comment by sarah872** [comment]
It works now with the new version!
However, I fail in saving the plot to pdf or png by using

```
pdf("plot.pdf")
vt_d3(canadaH, label=T,width = 900,height = 900, color_border = "#000000", size_border = "2px", legend = TRUE, legend_title = "Class", seed = 5)
dev.off()
```

Do you have an idea how to save it otherwise?

**Comment by alexkowa** [comment]
The function generates a htmlwidget and not a regular graph, so there are two options either
- use 'save as html' in Rstudio and then open it in Chrome and print to pdf
- or you can use the webshot package

**Comment by AaryaDeshpande** [comment]
TL;DR: my fix was converting the data frame using "as.data.frame" and passing that in to vt_input_from_df.

---
I had a similar issue when using this wonderful voronoiTreemap tool for a TidyTuesday dataset I was working to visualize.

I had a line `vt_data <- vt_input_from_df(tidy_tuesday_dataset)` that led to the error: 
```
vt_create_node(inp[1, 1]) : 
  is_scalar_character(total_lab) is not TRUE
``` 
This traceback error suggests that the `vt_input_from_df` function had within it some code called `vt_create_node` that didn't like something about the data frame I passed. Therefore, I investigated the `vt_input_from_df` function definition, searched for "vt_create_node," and found the full line: `n <- vt_create_node(inp[1, 1])`. For whatever this is doing, I suspected if `vt_create_node` isn't happy, then it's because of the input it's getting: `inp[1, 1]`. Since "inp" refers to the input dataframe (`function (inp, scaleToPerc = FALSE) ` is the function header), this means the function tried doing "[1,1]" to that dataframe and got something unexpected. But what exactly the unexpected thing I got, I was unsure. However, since I knew the example code from this documentation DID work, and it had no issue extracting "[1,1]" from the example ExampleGDP data frame, I decided to find out the output of ExampleGDP[1,1] and compare it with my tidy_tuesday_dataset[1,1]. Sure enough, there indeed was a difference:

ExampleGDP[1,1] returned the expected character:

<img width="107" alt="Image" src="https://github.com/user-attachments/assets/e57e328f-c0d6-4dde-bac2-6555a31b481b" />

However... tidy_tuesday_dataset[1,1] returned a tibble:

<img width="103" alt="Image" src="https://github.com/user-attachments/assets/f6861260-5b10-4325-8dce-19e2e0ace69c" />

I am unsure what was different about my dataframe to cause it to return a tibble after applying "[1,1]," as opposed to the ExampleGDP, which, after applying "[1,1]," actually returns the character output `[1] Total Levels: Total`. But there was something indeed different about my dataframe that made it output differently when doing "[1,1]" to it, so I looked up online for dataframe conversions, tried "as.data.frame(tidy_tuesday_dataset)," and it worked!

```
as.data.frame(tidy_tuesday_dataset) # Convert data to Voronoi format
vt_json <- vt_export_json(vt_data) # Export as JSON
vt_d3(vt_json) # Render Voronoi Treemap
```
NOTE: I am not an expert in R and am I'm still learning, so don't mind the overly trivial "explain like I'm five" explanation I wrote :)
